app/views/idv/shared/_document_capture.html.erb

```diff
@@ -10,6 +10,9 @@
   endpoint: api_verify_images_url,
   sp_name: sp_name,
   failure_to_proof_url: failure_to_proof_url,
+  front_image_upload_url: s3_presigned_url(keyname: "#{flow_session[:document_capture_session_uuid]}-front"),
```

Comment:
I'm not a huge fan of Rails helpers like this that get magically included, its hard to trace where these come from

A few ideas:
- Instead of a helper, we could make a service like `S3PresignedUrlBuilder`
- Generate the URLs, save as @-vars in the controller ex:

```ruby 
url_builder = S3PresignedUrlBuilder.new
@back_image_upload_url = url_builder.presign_url(keyname: "aaa")
@from_image_upload_url = url_builder.presign_url(keyname: "bbb")
```

Reply:
Heh, that's where I started, but then I ended up with enough in the helper I ended up abandoning it. :-P I'll go back...

Reply:
Ah, so the real reason I abandoned that was that there isn't a controller, per se, since this is a step in a flow so the "controller" is the flow state machine and I don't see an obvious way to set up instance variables.